stellargraph/layer/appnp.py

```diff
@@ -137,19 +137,17 @@ def call(self, inputs):
                 "Currently full-batch methods only support a batch dimension of one"
             )
 
-        # Remove singleton batch dimension
-        features = K.squeeze(features, 0)
-        propagated_features = K.squeeze(propagated_features, 0)
-
         # Propagate the node features
         A = As[0]
-        output = (1 - self.teleport_probability) * K.dot(
-            A, propagated_features
-        ) + self.teleport_probability * features
+        if K.is_sparse(A):
+            propagated_features = K.squeeze(propagated_features, 0)
```

Comment:
This could have a FIXME(#1222), since I think that this could be generalised in the same manner.

Reply:
@kieranricardo did you ignore this on purpose? (Not a problem if you don't think it applies, but a reply at least would be appreciated...)